include/contraction/contractionGraph.hpp

```diff
@@ -70,6 +73,46 @@ class contractionGraph :
         return --min_edge_id;
     }
 
+    /*!
+        @brief for C calls: to get the metric of a node, directly from the graph
+        @param [in] v vertex_descriptor
+        @return int64_t: the value of the metric for node v
+    */
+    double get_M(int64_t vertex_id) {
```

Comment:
Could you set a more explicit function name? `get_vertex_metric`?

Reply:
yes